model/riscv_sys_regs.sail

```diff
@@ -220,81 +219,80 @@ function get_mstatus_SXL(m : Mstatus) -> arch_xlen = {
   else m.bits[35 .. 34]
 }
 
-function set_mstatus_SXL(m : Mstatus, a : arch_xlen) -> Mstatus = {
-  if   xlen == 32
-  then m
-  else {
-    let m = vector_update_subrange(m.bits, 35, 34,  a);
-    Mk_Mstatus(m)
-  }
-}
-
 function get_mstatus_UXL(m : Mstatus) -> arch_xlen = {
   if   xlen == 32
   then arch_to_bits(RV32)
   else m.bits[33 .. 32]
 }
 
-function set_mstatus_UXL(m : Mstatus, a : arch_xlen) -> Mstatus = {
-  if   xlen == 32
-  then m
-  else {
-    let m = vector_update_subrange(m.bits, 33, 32,  a);
-    Mk_Mstatus(m)
-  }
+function get_mstatus_SD(m : Mstatus) -> bits(1) = {
+  if   xlen == 32 then m[SD_32]
+  else m[SD_64]
 }
 
-function legalize_mstatus(o : Mstatus, v : xlenbits) -> Mstatus = {
+function legalize_mstatus(o : Mstatus, v : bits(64)) -> Mstatus = {
   /*
    * Populate all defined fields using the bits of v, stripping anything
    * that does not have a matching bitfield entry. All bits above 32 are handled
    * explicitly later.
    */
-  // TODO: This method is error prone. We should instead explicitly copy the
-  // fields that are implemented. See legalize_mie() for an example.
-  let m : Mstatus = Mk_Mstatus(zero_extend(v[22 .. 7] @ 0b0 @ v[5 .. 5] @ 0b0 @ v[3 .. 3] @ 0b0 @ v[1 .. 1] @ 0b0));
+  let m : Mstatus = Mk_Mstatus(v);
 
+  // Set dirty bit to OR of other status bits.
+  let dirty = extStatus_of_bits(m[FS]) == Dirty |
+              extStatus_of_bits(m[XS]) == Dirty |
+              extStatus_of_bits(m[VS]) == Dirty;
+
+  [o with
+    SD_64 = if xlen == 64 then bool_to_bits(dirty) else o[SD_64],
+    SD_32 = if xlen == 32 then bool_to_bits(dirty) else o[SD_32],
+    MDT = m[MDT],
```

Comment:
I think we shouldn't add fields and make them writeable if we don't actually support them. I would probably just leave these fields out entirely for now. It's less confusing if we add them when we actually implement their functionality.